### ISSUE TITLE
For newest version SAILOR (v1.1.0)

### ISSUE THREAD

**Issue by XRCHAM** [YeoLab/sailor#6]
Hi there,

I notice the recent publication(https://www.nature.com/articles/s41592-021-01128-0) mentioned there is a new version for sailor (v1.1.0)"SAILOR calls for C-to-U edits". I would like to know whether it will update here recently or somewhere else?

Thanks!

**Comment by byee4** [comment]
Hi, this hasn't been wrapped up for release yet, but all of the scripts are updated to call C-U(T) candidates. You can use the '`ct: true`' flag, see the example below:

https://github.com/YeoLab/sailor/blob/master/example/example.json

**Comment by XRCHAM** [comment]
Hi, Thanks for your reply. When I paste the "ct: true" flag into the example yaml file, and run 
`./sailor-1.0.4` ce11_example.1.yaml 
I got the error: "Resolved '/opt/sailor/cwl/rnaediting2strands.workflow.cwl' to 'file:///opt/sailor/cwl/rnaediting2strands.workflow.cwl'
ce11_example.1.yaml:22:1: invalid field `ct`, expected one of: 'alpha', 'beta', 'dp', 'edge_mutation', 'edit_fraction', 'fwd_is_reverse', 'input_bam', 'junction_overhang', 'keep_all_edited', 'known_snp', 'min_variant_coverage', 'non_ag', 'reference', 'rev_is_reverse', 'reverse_stranded_library', 'single_end', 'skip_duplicate_removal'".

So would you mind explaining more details about how to integrate the "ct: true" flag into the sailor-1.0.4 package? 
Thanks!

**Comment by XRCHAM** [comment]
Besides, I am a newbie and have no idea instead how to directly use the example.json script.

**Comment by byee4** [comment]
Apologies for not wrapping this up as cleanly, but in the interim:
- Download and unzip the last stable version: [link](https://github.com/YeoLab/sailor/archive/refs/tags/v1.2.0.tar.gz)
- Untar and ensure that the following directories are in your $PATH: `CWL-SINGULARITY-pipeline-building-code/cwl`, `CWL-SINGULARITY-pipeline-building-code/py`
- Install the following softwares with conda: `python=2.7 samtools=1.3.1, bcftools=1.2, pysam, bedtools, pybedtools, pandas, numpy, scipy`.
- Install `cwltool` with pip: `pip install cwltool`
- Try running the cwl documents using the example yaml file: `cwltool cwl/wf_rnaediting2strands.cwl example/ce11_example.yaml` (this example is from an ADAR (A>G) dataset and while `ct: true` will work, you may not see any edits using the example file).

**Comment by gaze-abyss** [comment]
> Apologies for not wrapping this up as cleanly, but in the interim:
> 
> * Download and unzip the last stable version: [link](https://github.com/YeoLab/sailor/archive/refs/tags/v1.2.0.tar.gz)
> * Untar and ensure that the following directories are in your $PATH: `CWL-SINGULARITY-pipeline-building-code/cwl`, `CWL-SINGULARITY-pipeline-building-code/py`
> * Install the following softwares with conda: `python=2.7 samtools=1.3.1, bcftools=1.2, pysam, bedtools, pybedtools, pandas, numpy, scipy`.
> * Install `cwltool` with pip: `pip install cwltool`
> * Try running the cwl documents using the example yaml file: `cwltool cwl/wf_rnaediting2strands.cwl example/ce11_example.yaml` (this example is from an ADAR (A>G) dataset and while `ct: true` will work, you may not see any edits using the example file).

Thank you for your reply.
But when I ran the test data with this workflow, the following error occurred:

![image](https://user-images.githubusercontent.com/66149548/128802492-5557fe9b-4ad0-452d-b46f-62d6edd22184.png)

I'm sure the CWL-SINGULARITY-pipeline-building-code/cwl, CWL-SINGULARITY-pipeline-building-code/py are eligible.
I hope I can get your help.
Thank you.

**Comment by byee4** [comment]
Can you see if you can find `CWL-SINGULARITY-pipeline-building-code/py` in your $PATH? I think `split_strands.py` exists and is executable? If you can double check by: `echo $PATH` and `chmod +x CWL-SINGULARITY-pipeline-building-code/py/*.py`  since it seems like the error is related to not finding the program.

**Comment by gaze-abyss** [comment]
> Can you see if you can find `CWL-SINGULARITY-pipeline-building-code/py` in your $PATH? I think `split_strands.py` exists and is executable? If you can double check by: `echo $PATH` and `chmod +x CWL-SINGULARITY-pipeline-building-code/py/*.py` since it seems like the error is related to not finding the program.

Yes, that's what I find strange. The split_strands.py can be run separately, but the error above occurs when I execute the cwltool command line.

![image](https://user-images.githubusercontent.com/66149548/128963931-67e392e5-793b-441b-8768-5a69c6291d11.png)

**Comment by gaze-abyss** [comment]
> 很抱歉没有把它总结得那么干净，但在此期间：
> 
> * 下载并解压最新的稳定版本：[链接](https://github.com/YeoLab/sailor/archive/refs/tags/v1.2.0.tar.gz)
> * 解压并确保以下目录都在$ PATH： `CWL-SINGULARITY-pipeline-building-code/cwl`，`CWL-SINGULARITY-pipeline-building-code/py`
> * 使用 conda 安装以下软件：`python=2.7 samtools=1.3.1, bcftools=1.2, pysam, bedtools, pybedtools, pandas, numpy, scipy`.
> * `cwltool`使用 pip安装：`pip install cwltool`
> * 尝试使用示例 yaml 文件运行 cwl 文档：（`cwltool cwl/wf_rnaediting2strands.cwl example/ce11_example.yaml`此示例来自 ADAR (A>G) 数据集，虽然`ct: true`可以运行，但使用示例文件您可能看不到任何编辑）。



> Apologies for not wrapping this up as cleanly, but in the interim:
> 
> * Download and unzip the last stable version: [link](https://github.com/YeoLab/sailor/archive/refs/tags/v1.2.0.tar.gz)
> * Untar and ensure that the following directories are in your $PATH: `CWL-SINGULARITY-pipeline-building-code/cwl`, `CWL-SINGULARITY-pipeline-building-code/py`
> * Install the following softwares with conda: `python=2.7 samtools=1.3.1, bcftools=1.2, pysam, bedtools, pybedtools, pandas, numpy, scipy`.
> * Install `cwltool` with pip: `pip install cwltool`
> * Try running the cwl documents using the example yaml file: `cwltool cwl/wf_rnaediting2strands.cwl example/ce11_example.yaml` (this example is from an ADAR (A>G) dataset and while `ct: true` will work, you may not see any edits using the example file).

I want to confirm whether the software is running based on python=2.7?

**Comment by gaze-abyss** [comment]
> Apologies for not wrapping this up as cleanly, but in the interim:
> 
> * Download and unzip the last stable version: [link](https://github.com/YeoLab/sailor/archive/refs/tags/v1.2.0.tar.gz)
> * Untar and ensure that the following directories are in your $PATH: `CWL-SINGULARITY-pipeline-building-code/cwl`, `CWL-SINGULARITY-pipeline-building-code/py`
> * Install the following softwares with conda: `python=2.7 samtools=1.3.1, bcftools=1.2, pysam, bedtools, pybedtools, pandas, numpy, scipy`.
> * Install `cwltool` with pip: `pip install cwltool`
> * Try running the cwl documents using the example yaml file: `cwltool cwl/wf_rnaediting2strands.cwl example/ce11_example.yaml` (this example is from an ADAR (A>G) dataset and while `ct: true` will work, you may not see any edits using the example file).

I want to confirm whether the software is running based on python=2.7?  I don't have this problem when I use python=3.7 and I have another problem:

![image](https://user-images.githubusercontent.com/66149548/128985774-6f3186af-b20c-477f-bb04-f8f8b178148d.png)

**Comment by byee4** [comment]
The scripts may be python3 compatible (or nearly compatible aside from a few potential modifications), but I know that for this version, the environment we use is python2. "ifnotskip" should be in bin/, so if you're having trouble, can you do the same as what you did with py?

```
export PATH=/path/to/CWL-SINGULARITY-pipeline-building-code/bin/:$PATH
chmod +x /path/to/CWL-SINGULARITY-pipeline-building-code/bin/*
echo $PATH
which ifnotskip
```

**Comment by seven1112233** [comment]
Hi, I did what you described! well done!
But I got a strong confusion, what I want is C>U mutation, but the results have no record about it. How should I do?

**Comment by byee4** [comment]
Hi,

Make sure these scripts being used have this parameter (--ct) and that the flag is on:

For example: https://github.com/YeoLab/sailor/blob/5eb7864dd3622552dc95a927d3eb6ccf5f107745/bin/filter_variants.py#L369
https://github.com/YeoLab/sailor/blob/5eb7864dd3622552dc95a927d3eb6ccf5f107745/bin/filter_reads.py#L499

**Comment by seven1112233** [comment]
Thanks for your reply. Actually, I am sure that these scripts have being used and have this parameter (--ct) and that the flag is on in the *.cwl file as cwl/wf_rnaediting2strands.cwl(the default value of --ct is "false", I exchanged it with "true") . And my *.ranked.bed shows C > T, C>T,A and C>T,G mutation results. But what i want is  C>U mutation results.

**Comment by seven1112233** [comment]
Thank you very much! I get it!!! Your reply is very helpful!

**Comment by seven1112233** [comment]
Hi,
I also noticed that the recent publication(https://www.nature.com/articles/s41592-021-01128-0) mentioned  a combination of filters among which is (Poisson-based filter) and I have a question. Could I filter the edit sites only by  "score-based filter" with value 0.999? what is the impact with or without "Poisson-based filter"?
